internal/corazawaf/transaction.go

```diff
@@ -638,25 +638,41 @@ func (tx *Transaction) ExtractGetArguments(uri string) {
 
 // AddGetRequestArgument
 func (tx *Transaction) AddGetRequestArgument(key string, value string) {
-	// TODO implement ARGS value limit using ArgumentsLimit
+	if tx.checkArgumentLimit(tx.variables.argsGet) {
+		tx.debugLogger.Warn().Msg("skipping get request argument, over limit")
+		return
+	}
 	tx.variables.argsGet.Add(key, value)
 }
 
 // AddPostRequestArgument
 func (tx *Transaction) AddPostRequestArgument(key string, value string) {
-	// TODO implement ARGS value limit using ArgumentsLimit
+	if tx.checkArgumentLimit(tx.variables.argsPost) {
+		tx.debugLogger.Warn().Msg("skipping post request argument, over limit")
+		return
+	}
 	tx.variables.argsPost.Add(key, value)
 }
 
 // AddPathRequestArgument
 func (tx *Transaction) AddPathRequestArgument(key string, value string) {
-	// TODO implement ARGS value limit using ArgumentsLimit
+	if tx.checkArgumentLimit(tx.variables.argsPath) {
+		tx.debugLogger.Warn().Msg("skipping path request argument, over limit")
+		return
+	}
 	tx.variables.argsPath.Add(key, value)
 }
 
+func (tx *Transaction) checkArgumentLimit(c *collections.NamedCollection) bool {
+	return tx.WAF.ArgumentLimit != 0 && c.Len() >= tx.WAF.ArgumentLimit
+}
+
 // AddResponseArgument
 func (tx *Transaction) AddResponseArgument(key string, value string) {
-	// TODO implement ARGS value limit using ArgumentsLimit
+	if tx.WAF.ArgumentLimit != 0 && tx.variables.responseArgs.Len() >= tx.WAF.ArgumentLimit {
```

Comment:
Considering that the ArgumentLimit must be >= 0, does checking for `tx.WAF.ArgumentLimit != 0` make sense here?

Reply:
In directiveSecArgumentsLimit func , i check  limit is bigger than 0. Should i  check it again?
`func directiveSecArgumentsLimit(options *DirectiveOptions) error {
	limit, err := strconv.Atoi(options.Opts)
	if err != nil {
		return err
	}
	if limit <= 0 {
		return errors.New("argument limit should be bigger than 0")
	}`

Reply:
I should check tx.WAF.ArgumentLimit is bigger than 0, not equals 0.

Reply:
I'm just saying that we cannot set it to zero or negative. So the check doesn't make sense... imho...

Reply:
fixed

Reply:
[Here](https://github.com/corazawaf/coraza/pull/812/files#diff-ac03ba9c103ce13c145e2b00e0f7064decc5fdbaec3e05deafc491c54927d144R1010-R1012) you are already taking that into account.

Reply:
I deleted tx.WAF.ArgumentLimit != 0 to make sense